src/components/MultiHopTrade/MultiHopTrade.tsx

```diff
@@ -86,6 +99,44 @@ export const MultiHopTrade = (props: CardProps) => {
     })
   }, [assetSearch, setBuyAsset, supportedBuyAssets])
 
+  const totalProtocolFees = useMemo(() => {
+    if (!quoteData) return {}
+    return quoteData.steps.reduce<Record<AssetId, ProtocolFee>>((acc, step) => {
```

Comment:
This logic feels better placed in a helper, like we have for `getTotalNetworkFeeFiatPrecision`, for e.g.